tensorflow_probability/python/sts/dynamic_regression.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Dynamic Linear Regression model."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# Dependency imports
+import tensorflow as tf
+
+from tensorflow_probability.python import bijectors as tfb
+from tensorflow_probability.python import distributions as tfd
+from tensorflow_probability.python.internal import dtype_util
+
+from tensorflow_probability.python.sts.internal import util as sts_util
+from tensorflow_probability.python.sts.structural_time_series import Parameter
+from tensorflow_probability.python.sts.structural_time_series import StructuralTimeSeries
+
+
+def _num_features(design_matrix):
+  """Determine the number of features from the design matrix"""
+  if tf.compat.dimension_value(design_matrix.shape[-1]) is None:
+    return design_matrix.shape_tensor()[-1]
+  return design_matrix.shape[-1]
```

Comment:
you can replace this with

```
from tensorflow_probability.python.internal import prefer_static

# further down
num_features = prefer_static.shape(design_matrix)[-1]
```

Reply:
Oh nice, didn't see that. Fixed in 970eb39

As a slight aside, this code is repeated twice in [regression.py](https://github.com/tensorflow/probability/blob/master/tensorflow_probability/python/sts/regression.py):

``` python
if tf.compat.dimension_value(design_matrix.shape[-1]) is not None:
    num_features = design_matrix.shape[-1]
else:
    num_features = design_matrix.shape_tensor()[-1]
```

so might be worth making the same change there.

Reply:
good catch! I'm not sure we have a static shape utility that works with linearoperators (what that code is doing), but perhaps we should.